src/choleskyQR.jl

```diff
@@ -0,0 +1,141 @@
+function cholesky_jacobimatrix(w::Function, P::OrthogonalPolynomial)
+    !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    W = Symmetric(P \ (w.(axes(P,1)) .* P)) # Compute weight multiplication via Clenshaw
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+function cholesky_jacobimatrix(W::AbstractMatrix, P::OrthogonalPolynomial)
+    !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    !(W isa Symmetric) && error("Weight modification matrix must be symmetric.")
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+
+# The generated Jacobi operators are symmetric tridiagonal, so we store their data as two bands
+mutable struct CholeskyJacobiBands{T} <: AbstractCachedMatrix{T}
+    data::Matrix{T}
+    U::UpperTriangular
+    X::SymTridiagonal{T}
+    datasize::Int
+    array
+end
+
+# Computes the initial data for the Jacobi operator bands
+function CholeskyJacobiBands(W::Symmetric{T}, P::OrthogonalPolynomial) where T
+    U = cholesky(W).U
+    X = jacobimatrix(P)
+    dat = zeros(T,2,10)
+    for k in 1:10
+        dat[1,k] = (U * (X * (U \ [zeros(k-1); 1; zeros(∞)])))[k]
+        dat[2,k] = (U * (X * (U \ [zeros(k); 1; zeros(∞)])))[k]
+    end
+    return CholeskyJacobiBands{T}(dat, U, X, 10, dat)
+end
+
+size(::CholeskyJacobiBands) = (2,ℵ₀) # Stored as two infinite bands
+
+# Resize and filling functions for cached implementation
+function resizedata!(K::CholeskyJacobiBands, nm::Integer)
+    νμ = K.datasize
+    if nm > νμ
+        olddata = copy(K.data)
+        K.data = similar(K.data, 2, nm)
+        K.data[axes(olddata)...] = olddata
+        inds = νμ:nm
+        cache_filldata!(K, inds)
+        K.datasize = nm
+    end
+    K
+end
+function cache_filldata!(J::CholeskyJacobiBands, inds::UnitRange{Int})
+    for k in inds
+        J.data[1,k] = (J.U * (J.X * (J.U \ [zeros(k-1); 1; zeros(∞)])))[k]
```

Comment:
This doesn't have the right complexity: we can recover the entries in $O(1)$ operations whilst this is $O(k)$.

Reply:
How does one compute $U X U^{-1}$ in $O(1)$ complexity? That would be a big improvement obviously.

Reply:
ah, I think I get it now. expand by all new inds instead of one by one. should be straightforward to change

Reply:
Write the enty as `y' * U^{-1} e_j ` where `y = (X'U'e_k)` has only the non-zero entries `y[k:k+u+1]` where `u` is the upper-bandwidth of `U`. That is, we only need `(U \ e_j)[max(j,k):min(k+u+1,j)]` (`j = k-1:k+1` as other entries are zero)

Reply:
I think it's `y[k-1:k+u+1]` that is non-zero but I understand yeah